### ISSUE TITLE
Unable to set page overlap ON as a default option

### ISSUE THREAD

**Issue by offset-torque** [koreader/koreader#7757]
- PAGE OVERLAP is a mandatory feature while using CONTINUOUS MODE otherwise last line is unreadable most of the time
- CONTINUOUS MODE can be set as a default so users should be able to set PAGE OVERLAP as a default too

Curiously we can set **PAGE OVERLAP STYLE (arrow/gray out)** as a default.

So in this hierarchy middle one is missing:

`CONTINUOUS MODE (can be set as default) > PAGE  OVERLAP (nope) > PAGE OVERLAP STYLE (can be set as default)`

**Comment by NiLuJe** [comment]
Trickier because it's a dynamic setting (which gets disabled in paged mode or if the zoom mode makes it useless).

**Comment by offset-torque** [comment]
Default options are read and set for current document only on first file opening. 
Then any changes to settings written to document metadata.lua if I am correct. 
This line controls the option I guess:
`["show_overlap_enable"]`

So what is wrong with this flow (I am trying wrap my head around):
1. Open all documents with `["show_overlap_enable"] = true` because user set it as a DEFAULT
2. If user clicks PAGED MODE or uses zoom, disable it and set `["show_overlap_enable"] = false`

Isn't this process same as:
1. Open all documents with Times New Roman (default option) 
2. Until user changes to Verdana (updated in metadata.lua)

Also in this case, if a user wants to open **ALL** documents in CONTINUOUS MODE they will definitely want PAGE OVERLAP set as **ON** too. I don't think there are any users that enjoys their last lines cut randomly. So PAGE OVERLAP can even be directly tied to CONTINUOUS MODE as a default ON. 

Analogy might be like "ice cream and cone". They are two separate items but if shop only sells in cones then cones becomes a default. If I say "I want ice cream" (CONTINUOUS MODE) that automatically includes a cone (PAGE OVERLAP) too. Of course I can have only ice cream but it is hard to use without a cone so no one actually chooses this.

Of course there are users that might want to **SCROLL** the document in CONTINUOUS MODE instead of page turning but PAGE OVERLAP is not relevant to them so being on as DEFAULT won't do any harm to them. They won't see any overlap because they are scrolling.

Sorry for long text but I wanted to be clear.

**Comment by NiLuJe** [comment]
Okay, brain works better when I'm not half asleep.

There's already a global default for this, but it's a legacy one (e.g., Advanced settings => `DSHOWOVERLAP`).

**Comment by offset-torque** [comment]
> There's already a global default for this, but it's a legacy one (e.g., Advanced settings => DSHOWOVERLAP).

Thank you NiLuJe. This is the exact setting I was asking for. So it automatically closes the issue because this option exists. (Unless someone decides to remove it from legacy options)

But I think we have some kind of documentation problem here. This is the second topic that I learned there exists a solution after opening an issue which I couldn't find reading the wiki. KOReader is a complex program that needs an in depth manual I think. Like some kind of reference PDF manual instead of clickable individual wiki pages. I would be happy to help about that.

**Comment by Frenzie** [comment]
See https://github.com/koreader/koreader/issues/1371 for discussion.

**Comment by offset-torque** [comment]
I checked the linked thread @Frenzie . That is a looong thread about the technicalities of exporting github wiki. I was specifically talking about the content of manual:

I find it funny that everyone is trying to produce the best solution for exporting wiki where %90 of pages in it are 2-3 years old (24-36 releases ?!). SCREENSHOTS are 4 years old, none of the icons even look like that. BASIC USAGE category is missing and says TODO.

**For the current example:**
The info that I got from @NiLuJe  in this thread. Suppose that I want to add this to documentation so people can read it there and don't open issues like I did. Logical place would be under **CONTINUOUS MODE** heading right? Whoops doesn't exist. So in order to create that heading I should also create **PAGE RENDERING MODES** section. Hmmm, where does **PAGE RENDERING MODES** belong, maybe under **LOWER TOOLBAR** section. Nope, that page doesn't exist too. There is no **USER INTERFACE** entry even. Maybe I can put under **FAQ** or **"How can I ..."** pages. Nyet. They don't exist too. So should I go ahead and create all these pages ?

Maybe github is not a suitable place to prepare this type of document. I can understand the appeal of "everybody can edit easily" but from the update frequency it is obvious that no one is interested in it.

I have published a few books (on paper), edited some and used digital publishing software like QuarkXpress and Scribus so I can say I know a few things about creating user facing documents. I personally think that a collaborative word document that is regularly exported to PDF will be much more easier and it would look more professional and helpful. I can volunteer for this. But if YOU (whoever is making these kind of decisions) say **"Nah, we are ok like this"** I have nothing to do.

**Comment by pazos** [comment]
> I can volunteer for this. But if YOU (whoever is making these kind of decisions) say "Nah, we are ok like this" I have nothing to do.

This app is a colaborative effort made by volunteers. Contributing documentation is **very welcome** and one of the ways (the generic) you can help. See https://github.com/koreader/koreader#support

Of course, our best case scenario is: an user discovers how to do something which is undocumented, writes documentation to help others to do the same.

But if you're willing to do *more* than documenting specific features and want to do the humongous task of writting an user manual you're **VERY WELCOME** 

> I have published a few books (on paper), edited some and used digital publishing software like QuarkXpress and Scribus so I can say I know a few things about creating user facing documents. I personally think that a collaborative word document that is regularly exported to PDF will be much more easier and it would look more professional and helpful.

The format itself isn't important for the moment. The content is. Once there's some content written we can discuss which format is better. There's some outstanding tools out there, like pandoc, than can generate multiple targets (pdf, epub, groff..) from a single source.

In short: **YES PLEASE**, Do write content.

**Comment by NiLuJe** [comment]
What @pazos said ;).

As for this in particular, the info is there, it's just buried in something you only find if you already know where it's buried (kinda ;p): https://github.com/koreader/koreader/wiki/Change-defaults

**Comment by mergen3107** [comment]
So. back to overlap question. I have a suggestion to  introduce graying out for `Page Overlap`. Here is how I think it should work:
1) Whenever it is not grayed out, let the user set default option for `Page overlap`, on or off.
2) However, the actual application of this setting will depend on whether this whole setting is grayed out or not, which depends on context.

For example, the user sets it on by default. Opens a document, which he knows he will need that feature (paged mode, fit to width, long pages in pdf). As soon as any option making the page overlap conflicting/useless is selected, make the page overlap grayed out and inactive, but preserve its tick. So when such option is deselected, the previous page overlap will be back on.

So essentially I suggest to add an intermediate layer of applying this setting to separate user's default choice from the actual application of the setting:
- Move all current conditions that invalidate the page overlap from on/off to graying out;
- create new sub-option of user's on/off.

**Comment by hius07** [comment]
I would propose as follows:
introduce new G_reader_settings("show_overlap_enable") instead of DSHOWOVERLAP (default for new books).
Main menu `Style - Page overlap` always enabled.
In the submenu:
Tap `[  ] Page overlap` - if can be enabled, check and enable / if not, show infomessage "Page overlap cannot be enabled for current view mode".
Long-press `[  ] Page overlap` - toggles G_reader_settings("show_overlap_enable") (with ★).

I haven't digged deep enough yet to understand if duplicated menu code in readerpaging and readerrolling can be easily separated, similar to https://github.com/koreader/koreader/pull/8022.

**Comment by Frenzie** [comment]
Works for me. :+1:

**Comment by hius07** [comment]
I'm finishing new Page overlap menu and encountered a possible bug.
Enable overlap, then do not disable overlap and switch to the view mode not supporting overlap.
The dim area (grayed or arrow) is not cleared and continues to be shown on every page (while there is no actual overlap).
It seems to me it's a bug, but I am not an expirienced overlap user so please advise.
Is it by intention or should I fix it?

**Comment by Frenzie** [comment]
The view mode not supporting overlap? Not sure what you mean, but it sounds like it supports it and the menu shouldn't be grayed out.

**Comment by hius07** [comment]
Open epub, select `continuous` view mode in the bottom menu, enable overlap.
Select `page` view mode, overlap menu is grayed, there is no overlap.
Dim area remains grayed or arrowed.

**Comment by Frenzie** [comment]
> Open epub

Huh, who knew EPUB also had an overlap feature. Tbh continuous in EPUB is just something that's kind of… there, useful for multi-page selection though.

So yeah, clearly overlap shouldn't apply in paged mode. :-)

**Comment by hius07** [comment]
> clearly overlap shouldn't apply in paged mode

It is not a real overlap (there are no repeating lines), just dimmed area on every page. I'll fix it.

**Comment by Frenzie** [comment]
Yes, that's what I'm saying. Clearly nothing should be grayed out. ^_^

**Comment by poire-z** [comment]
Yep, it's a bug. Also happens when overlap set to arrow.
This draws the various overlap ways:
https://github.com/koreader/koreader/blob/cf1248393cad8e29a6fbd00ea9232235d2b36d46/frontend/apps/reader/modules/readerview.lua#L181-L193
But we only clear view.dim_area when disabling it via the menu:
https://github.com/koreader/koreader/blob/cf1248393cad8e29a6fbd00ea9232235d2b36d46/frontend/apps/reader/modules/readerrolling.lua#L408-L419

So, ReaderView should better check if overlap should apply or not, or somewhere else we should also view.dim_area.clear().

Don't expect all this to be pretty, I remember I did some small overlap fixes, but they were hacks over hacks... And ReaderView is not pretty.

**Comment by hius07** [comment]
> Yep, it's a bug.

Thank you, I've digged into it and remember these pieces of code by heart)
